### PR TITLE
Adding copying and moving constructor for Concatenate Layer

### PR DESCRIPTION
**Pull request by rxng8**
Reference to issue #2625 . I also fixed minor description error of a parameter.

### CODE REVIEW

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 1]
```diff
@@ -153,6 +153,56 @@ TEST_CASE("CheckCopyMovingVanillaNetworkTest", "[FeedForwardNetworkTest]")
   CheckMoveFunction<>(model1, trainData, trainLabels, 1);
 }
 
+/**
+ * Concatenate layer constructor test.
+ */
+TEST_CASE("CheckCopyMovingConcatenateTest", "[FeedForwardNetworkTest]")
+{
+  // Create training input by 5x5 matrix
```
Comment (kartikdutt18) [inline src/mlpack/tests/feedforward_network_test.cpp:161]: Hey @rxng8,
Could you also add a stop at the end sentences. Thanks.

Comment (rxng8) [reply src/mlpack/tests/feedforward_network_test.cpp]: Thank you for reminding me! Resolved!

**Comment by rxng8** [comment]
@kartikdutt18 just want to let you know I have resolved the review.

File: src/mlpack/methods/ann/layer/concatenate_impl.hpp [hunk 2]
```diff
@@ -25,6 +25,56 @@ Concatenate<InputDataType, OutputDataType>::Concatenate()
...
+Concatenate<InputDataType, OutputDataType>::
+operator=(const Concatenate& layer)
+{
+  if (this != &layer) 
+  {
+    inRows = layer.inRows;
+    weights = layer.weights;
+    delta = layer.delta;
+    concat = layer.concat;
+  }
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/layer/concatenate_impl.hpp:59]:
```suggestion
  }

```
:)

**Review by kartikdutt18 — COMMENTED** [review]
This looks good to me. Maybe some one could take a closer look at it too. Thanks for the contribution.

**Review by rcurtin — APPROVED** [review]
Yeah, this looks good to me too.  Thanks @rxng8! :+1:

**Comment by rxng8** [comment]
@rcurtin @kartikdutt18 I have merged this branch with the latest branch and tested it.

**Comment by kartikdutt18** [comment]
Hey @rxng8,
Can you also fix the static error in Concatenate layer. It requires initialization of parameters in default constructor as done in other layers. It should be pretty easy to fix.
Thanks.

**Review by kartikdutt18 — APPROVED** [review]
Great, Once the build comes back green, I will merge it. Thanks a lot for working on this PR and following up on the comments.

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 3]
```diff
@@ -241,6 +241,56 @@ TEST_CASE("CheckCopyMovingNoisyLinearTest", "[FeedForwardNetworkTest]")
...
+
+  // Add concatenate layer to the current network.
+  model1->Add(concatLayer);
+  model1->Add<Linear<> >(10, 5);
+  model1->Add<LogSoftMax<>>();
+
+  // Check whether copy constructor is working or not.
+  CheckCopyFunction<>(model1, input, output, 1);
+
+  // check moving constructor.
```
Comment (zoq) [inline src/mlpack/tests/feedforward_network_test.cpp:274]:
```suggestion
  // Check moving constructor.
```

Minor style fix.

File: src/mlpack/methods/ann/layer/concatenate.hpp [hunk 4]
```diff
@@ -82,7 +94,7 @@ class Concatenate
 
   //! Get the concat matrix.
   OutputDataType const& Concat() const { return concat; }
-  //! Modify the delta.
+  //! Modify the concat.
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/concatenate.hpp:97]: Nice catch!

**Comment by rxng8** [comment]
I fixed the static code check error. Thanks!

**Comment by kartikdutt18** [comment]
Thanks again for working on this.

**Comment by rxng8** [comment]
sure! Peace!

**Pull request merged**